components/common/src/templating/hooks.rs

```diff
@@ -315,6 +320,61 @@ pub trait Hook: fmt::Debug + Sized + Send {
     fn stderr_log_path(&self) -> &Path;
 }
 
+#[async_trait::async_trait]
+pub trait HookExt: Hook<ExitValue = ExitStatus> + Sync {
```

Comment:
The naming and typing of this trait doesn't clearly capture that it's for install and uninstall hooks only. It'd be nice to make that more obvious (though I admittedly am not sure what a better name for the trait and function would be).

Reply:
Agreed. I struggled around the naming of this. There is no reason this has to apply _only_ to the `install` and `uninstall` hooks so I left the naming very general.

The method and frequency of when the `install` and `uninstall` hooks are run may provide an axis of differentiation. What if I called the trait one of the following: `OneOffHookExt`, `AuxiliaryHookExt`, `PackageMaintenceHookExt`...?

Ultimately, I think the `Hook` trait could use some cleanup. There is a lot of duplicated functionality when calling the hooks that should be moved into the trait itself and the `handle_exit` functionality seems unnecessarily general. There is really only a few different kinds of exit types and this would probably be better handled with different run functions.

Reply:
I agree that the `Hook` trait needs a lot of love :frowning_face: 

The idea of using method/frequency of hook execution as an axis of differentiation is a great one. Of the suggestions given, I lean toward `PackageMaintenanceHookExt` as it's more descriptive and seems to capture something core to the whole idea of having `install` and `uninstall` hooks. Making that name change, along with a nice documentation comment, would be a nice addition :+1:

Reply:
Thanks! Addressed in [19aa2c](https://github.com/habitat-sh/habitat/pull/7782/commits/19aa2c96080261d223409866295e7bb56427046a)